addons/mail/static/src/discuss/web/avatar_card/avatar_card_popover.js

```diff
@@ -77,6 +139,43 @@ export class AvatarCardPopover extends Component {
         this.props.close();
     }
 
+    onClickRemove() {
+        this.dialog.add(ConfirmationDialog, {
+            body: _t('Do you want to remove "%(member_name)s" from this channel?', {
+                member_name: this.props.channelMember?.name,
+            }),
+            cancel: () => {},
+            confirm: () => {
+                rpc("/discuss/channel/remove_member", {
+                    member_id: this.props.channelMember.id,
+                });
+                this.props.close();
+            },
+        });
+    }
+
+    setChannelRole(role) {
+        if (
+            !this.store.self_user?.is_admin &&
+            (this.props.channelMember.threadAsSelf || role === "owner")
+        ) {
+            this.dialog.add(ConfirmationDialog, {
+                body: this.props.channelMember.threadAsSelf
+                    ? _t(
+                          "Do you want to remove owner from yourself? You will no longer have full control over the channel and its settings."
+                      )
+                    : _t(
+                          'Do you want to set "%(member_name)s" as the owner? This means that the member will have full control over the channel and its settings.\n\nThis action cannot be reverted.',
+                          { member_name: this.props.channelMember?.name }
```

Comment:
Inconsistent guard of channelMember across the various functions. If it's not guarded line 163, a guard is definitely not needed here.

Same in several other places. We probably never need the guard in these methods, as it would make no sense to do any of that if we don't have a member.

Reply:
This is inside an action that is only visible if we have a member in the first place.

Yes for the getter we need to keep them if we don't have a top-level condition for member existing before reaching this code.

Reply:
yes I've removed this one. I was mentioning the guards in the getters. As reference to `We probably never need the guard in these methods`, thought you were talking all the mothods in the file.